src/orange/orangeinp/CsgTypes.hh

```diff
@@ -146,6 +148,67 @@ inline constexpr bool is_boolean_node(Node const& n)
     return std::holds_alternative<True>(n) || std::holds_alternative<False>(n);
 }
 
+//---------------------------------------------------------------------------//
+// INTERFACE TYPES
+//---------------------------------------------------------------------------//
+/*!
+ * Polygon bound by top/bottom z segments, one of which may be a single point.
+ *
+ * Thus, valid polygons are:
+ *
+ * 1) quadrilaterals with two segments parallel to the z axis,
+ * 2) triangles with one segment parallel to the z axis.
+ */
+class SpecialTrapezoid
```

Comment:
This class is arguable large enough to test on it's own, but it is tested implicitly via `RevolvedSpecialTrapzoid`